src/Http/RequestMatcher/QuerystringRequestMatcher.php

```diff
@@ -12,18 +12,23 @@
 namespace FOS\HttpCacheBundle\Http\RequestMatcher;
 
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\RequestMatcher as SymfonyRequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
 
 /**
  * Extend the Symfony RequestMatcher class to support query string matching.
  */
-class QuerystringRequestMatcher extends SymfonyRequestMatcher
+class QuerystringRequestMatcher implements RequestMatcherInterface
 {
     /**
      * @var string Regular expression to match the query string part of the request url
      */
     private $queryString;
 
+    public function __construct($queryString = null)
```

Comment:
when we add this to the constructor, i'd like to remove the setter for queryString.